### PR TITLE
Update CHANGELOG.md

### PR DESCRIPTION
**Pull request by trodge**
Add 5.7.0 notes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,31 @@
-## 5.7.0 (Unreleased)
+## 5.7.0 (Nov 20, 2023)
```
Comment (zli82016) [inline CHANGELOG.md:1]: At top `## 5.8.0 (Unreleased)` is needed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,31 @@
...
+
+DEPRECATIONS:
+* gkehub: deprecated `config_management.binauthz` in `google_gke_hub_feature_membership` ([#16536](https://github.com/hashicorp/terraform-provider-google/pull/16536))
```
Comment (zli82016) [inline CHANGELOG.md:4]: Looks like this PR should be targeted for TPG instead of TPGB

File: CHANGELOG.md (hunk 1 above)
Comment (trodge) [reply CHANGELOG.md:3]: Should that be in the main branch at this point? I have separately raised https://github.com/hashicorp/terraform-provider-google-beta/pull/6682 for the `release-5.8.0` branch.

Comment (zli82016) [reply CHANGELOG.md:3]: It is needed based on the doc https://github.com/hashicorp/terraform-provider-google/wiki/Release-Process#on-monday

Comment (zli82016) [reply CHANGELOG.md:3]: I think it is better to add it to avoid confusion in the future.

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**